modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -154,6 +154,32 @@ class SoftMaxSubgraph : public Subgraph
     int axis;
 };
 
+class FuseGatherCastSubgraph : public Subgraph
+{
+public:
+    FuseGatherCastSubgraph()
+    {
+        int input = addNodeToMatch("");
+        int index = addNodeToMatch("Constant");
+        int gather = addNodeToMatch("Gather", input, index);
+        addNodeToMatch("Cast", gather);
+        setFusedNode("Gather", input, index);
+    }
+};
+
+class FuseMulCastSubgraph : public Subgraph
```

Comment:
Just `MulCastSubgraph` would be OK

Reply:
So should I remove changes related to `GatherCastSubgraph `and keep changes for `MulCastSubgraph` ?